### PR TITLE
lesson 6: feat: function palindrome, function findUniqueCountries don…

### PR DESCRIPTION
**Pull request by LudmilaZhuleha**
…e. Functions covered with tests

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1,31 +1,68 @@
...
+const findUniqueCountries = (array) => {
+  const arrayUnique = array.reduce((acc, item) => {
+    acc[item.country] = acc[item.country] || [];
+    acc[item.country].push(item.city);
+    return acc;
+  }, {});
+  return arrayUnique;
 };
-const data = [
+const hotels = [
```
Comment (Marinarium) [inline src/main.js]: hotels is a really more appropriate name =) it's a good refactoring
I recommend to keep such an information in a file with some moke data
For example you can create a folder "mocks" for such a type of data

File: src/main.js [hunk 2]
```diff
@@ -1,31 +1,68 @@
...
+console.log(findUniqueCountries(hotels));
+
+//Create a function getCalendarMonth, which receives parameters:
+//daysInMonth =30,
+//daysInWeek =7,
+//weeksInMonth=5,
+//and any day in week from 0 to 6 to start the month from.
+//The function should return the array of arrays illustrating calendar month.
+
+const getCalendarMonth = (weeksInMonth, daysInWeek, dayOfWeek) => {
```
Comment (Marinarium) [inline src/main.js]: You should use 
`daysInMonth`**-** _you've used_ `weeksInMonth`
`daysInWeek`**+**
`dayOfWeek`**+**

please use arguments according to your task

File: src/main.js [hunk 3]
```diff
@@ -1,31 +1,68 @@
...
+      if (counter === 31) counter = 1;
+      arr[i][j] = 1;
+      arr[i][j] = counter++;
+    }
+  }
+  return arr;
+};
+console.log(getCalendarMonth(5, 7, 1));
+
+export { findUniqueCountries, palindrome, getCalendarMonth };
```
Comment (Marinarium) [inline src/main.js:62]: where is import? if your functions aren't imported, don't add export
**yagni**
image

image

File: src/main.js [hunk 4]
```diff
@@ -1,31 +1,68 @@
...
+//Create function checking palindrome string for true/false
+const palindrome = (str) => str === str.split('').reverse().join('');
 
-const reversedStr = (str, separator, joiner) => {
-  return str.split(separator).reverse().join(joiner);
-};
-reversedStr('2020-11-26', '-', '.');
-
-//Create a function that searches the array for string input and console.log the found string city, country, hotel
+console.log(palindrome('molom'));
```
Comment (Marinarium) [inline src/main.js:7]: **were is the 6th task?**
Поиск объектов размещения:
дан массив;
напишите функцию поиска, которая будет принимать строку;
по полученной строке найдите все совпадения в массиве по любому из полей;
верните масиив строк в формате: страна, город, отель;
зная как работать с массивами, сократите вашу функцию избавившись от цикла for.

**Review by Marinarium — CHANGES_REQUESTED** [review]

File: src/main.js (hunk 3 above)
Comment (LudmilaZhuleha) [reply src/main.js:62]: Functions are imported in test.js files separately. Please check.

File: src/main.js (hunk 4 above)
Comment (LudmilaZhuleha) [reply src/main.js:7]: This task was done in Lesson5 without for loop. If necessary, I can add it again in lesson6.

File: src/main.js (hunk 1 above)
Comment (LudmilaZhuleha) [reply src/main.js]: New data folder added for variables.

File: src/main.js (hunk 2 above)
Comment (LudmilaZhuleha) [reply src/main.js]: Arguments refactored.

File: src/main.js (hunk 4 above)
Comment (LudmilaZhuleha) [reply src/main.js:7]: For 6th task function added.

**Pull request merged**